### PR TITLE
enhancement(lua transform): Allow traces to be passed through

### PR DESCRIPTION
**Pull request by jszwedko**
Credit to `ay#7001` on Discord for the patch. Opening the PR on their behalf.

Fixes: https://github.com/vectordotdev/vector/issues/17412

Signed-off-by: Jesse Szwedko <jesse.szwedko@datadoghq.com>

### CODE REVIEW

File: lib/vector-core/src/event/lua/trace.rs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    #![allow(clippy::wrong_self_convention)] // this trait is defined by mlua
+    fn to_lua(self, lua: &'a Lua) -> LuaResult<LuaValue> {
+        let (value, _metadata) = self.into_parts();
+        value.to_lua(lua)
+    }
+}
+
+impl<'a> FromLua<'a> for TraceEvent {
+    fn from_lua(lua_value: LuaValue<'a>, lua: &'a Lua) -> LuaResult<Self> {
+        //let value = Value::from_lua(lua_value, lua)?;
```
Comment (spencergilbert) [inline lib/vector-core/src/event/lua/trace.rs:15]:
```suggestion
```
?

**Review by bruceg — COMMENTED** [review]
I agree with the other nits raised, but otherwise this LGTM. Also `cargo fmt`.

**Review by lukesteensen — COMMENTED** [review]
My only hesitation here is that it effectively exposes our (presumably temporary) reuse of `LogEvent` as the representation of traces in a way that would make any change potentially breaking for user configs.

**Comment by jszwedko** [comment]
Closing since there is some contention about exposing the structure of traces.

**Pull request closed without merging**